elastic/apm/index-template.json

```diff
@@ -0,0 +1,9 @@
+{
+    "data_stream": {},
+    "index_patterns": ["traces-apm*", "logs-apm*", "metrics-apm*"],
+    "_meta": {
+        "description": "Placeholder template that'd be used when the x-pack-apm car is not used, e.g. in test mode"
+    },
+    "composed_of": ["common-apm@custom"],
+    "priority": 1
+}
```

Comment:
As mentioned above, I don't think the index-template should be specificed. This is overwriting the default index templates, and we wanted to rely on them for the settings, mappings and processors.

Reply:
This index template is needed as a placeholder to apply replica settings when there's no index template installed by ES plug-in, for example, in test mode(it does not install the ES Plugin for apm).

Also, this template is named differently to the index templates for APM so it won't be overwritten but co-existing with them. The thing we need to be careful is the index pattern collision. If we run the race against the cluster with ES Plug-in enabled(using `x-pack-apm` car), this placeholder template has a lower [priority](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/apm-data/src/main/resources/index-templates/traces-apm%40template.yaml#L4) than the one from the plug-in so the this index template will be ignored according to the [doc](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-templates.html) (_Avoid index pattern collisions_ section)

Reply:
ah, that clarifies things, I wasn't aware that the ES plugin is not installed for tests;
in that case I agree with the solution you created, and take back my comment to rename the component templates; with your solution the `replica:0` would only be applied in tests, but not when actually performing the rally track -> seems to be the right choice.